src/test/unit/algorithms/SpatialPoolerTest.cpp

```diff
@@ -2222,6 +2222,23 @@ namespace {
     EXPECT_EQ(0, countNonzero(activeColumns));
   }
 
+
+  TEST(SpatialPoolerTest, testSameOutputForSameInputNoLearningNoBoosting)
+  {
+    const UInt inputSize = 10;
+    const UInt nColumns = 20;
+    SpatialPooler sp;
+    sp.initialize({inputSize}, {nColumns});
+    sp.setBoostStrength(0);
```

Comment:
I'd suggest some explanatory comment here: 
"This test verifies possible issue where SP with 0-boosting and learning disabled produces different (random) output, compared to the python SP. "
and add
bool learning = False;